src/middleware/isAuthenticated.js

```diff
@@ -0,0 +1,4 @@
+const passport = require('passport')
+
+// access user info from req.user in follow-up function
+module.exports = () => passport.authenticate('jwt', { session: false })
```

Comment:
nice work @Yellow-May. this is just my curiosity nothing else. wanted to know the reason behind us having to put this in a function instead of just

```js
   module.exports = passport.authenticate('jwt', { session: false })
```
I'm thinking it will cause an issue this way but just want to be sure

Reply:
@jakazzy stated in the issue to have it as middleware function so it can be used in all cases without having to rewrite the same line of code repeatedly

Reply:
There won't be any issues
Since all it does it check if a user is authenticated and moves to the next function, if the user isn't authenticated, it just sends back and error

There are no other major functionalities.

The plus is just the reusability

Reply:
yea @Yellow-May i get that part, my question was mainly around the difference between

```js
module.exports = () => passport.authenticate('jwt', { session: false })
```

and 

```js
module.exports = passport.authenticate('jwt', { session: false })
```

One in a function and one directly assigned.

Reply:
i choose the function rather than just assigning it because I thought maybe there might be need to include conditions

Let's say, some middleware will need a different config than others, or something of the sort

Reply:
i choose the function rather than just assigning it because I thought maybe there might be need to include conditions

Let's say, some middleware will need a different config than others, or something of the sort.

What do you think?

Reply:
What do you think?

Reply:
> i choose the function rather than just assigning it because I thought maybe there might be need to include conditions
> 
> Let's say, some middleware will need a different config than others, or something of the sort

o okay I see, understood

Reply:
> What do you think?

Yea i think it makes sense. Thanks for the explanation 💯